nomad/structs/config/workload_id.go

```diff
@@ -0,0 +1,88 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package config
+
+import (
+	"slices"
+
+	"github.com/hashicorp/nomad/helper/pointer"
+)
+
+// WorkloadIdentityConfig is the agent configuraion block used to define
+// default workload identities.
+//
+// This based on the WorkloadIdentity struct from nomad/structs/workload_id.go
+// and may need to be kept in sync.
+type WorkloadIdentityConfig struct {
+	// Name is used to identity the workload identity. It is not expected to be
+	// set by users, but may have a default value.
+	Name string `mapstructure:"-" json:"-"`
+
+	// Audience is the valid recipients for this identity (the "aud" JWT claim)
+	// and defaults to the identity's name.
+	Audience []string `mapstructure:"aud"`
+
+	// Env injects the Workload Identity into the Task's environment if
+	// set.
+	Env *bool `mapstructure:"env"`
+
+	// File writes the Workload Identity into the Task's secrets directory
+	// if set.
+	File *bool `mapstructure:"file"`
+}
+
+func (wi *WorkloadIdentityConfig) Copy() *WorkloadIdentityConfig {
+	if wi == nil {
+		return nil
+	}
+	nwi := new(WorkloadIdentityConfig)
+	*nwi = *wi
+	nwi.Audience = slices.Clone(wi.Audience)
+
+	if wi.Env != nil {
+		nwi.Env = pointer.Of(*wi.Env)
+	}
+	if wi.File != nil {
+		nwi.File = pointer.Of(*wi.File)
+	}
+
+	return nwi
+}
+
+func (wi *WorkloadIdentityConfig) Equal(other *WorkloadIdentityConfig) bool {
+	if wi == nil || other == nil {
+		return wi == other
+	}
+
+	if wi.Name != other.Name {
+		return false
+	}
+
+	if !slices.Equal(wi.Audience, other.Audience) {
+		return false
+	}
+
+	if !pointer.Eq(wi.Env, other.Env) {
+		return false
+	}
+
+	if !pointer.Eq(wi.File, other.File) {
+		return false
+	}
+
+	return true
+}
+
+func (wi *WorkloadIdentityConfig) Merge(other *WorkloadIdentityConfig) *WorkloadIdentityConfig {
```

Comment:
The `structs` package usually merge things in-place, but on `config` the preference is to return a new struct.